### PR TITLE
vcsim: Implement SearchDatastoreSubFolders_Task

### LINKED ISSUE #2287: Need SearchDatastoreSubFolders_Task in VC simulator
We have two unit tests commented out due to the lack of this method in the VC simulator, in `/lib/install/management/delete_test.go` at the bottom of `TestDelete` you will find this:
```
		//	testDeleteVCH(validator, conf, t)
		// FIXME: ServerFaultCode: HostDatastoreBrowser:hostdatastorebrowser-34 does not implement: SearchDatastoreSubFolders_Task
		// testDeleteDatastoreFiles(validator, t)
```

`SearchDatastoreSubFolders_Task` needs to be implemented in vcsim so that `testDeleteVCH` and `testDeleteDatastoreFolders` can be re-enabled. 

This issue can be closed once those two tests are re-enabled.

### PR DESCRIPTION
**Pull request by dougm**
Fixes #2287

### CODE REVIEW

**Comment by emlin** [comment]
lgtm

File: lib/install/management/delete_test.go [hunk 1]
```diff
@@ -90,8 +90,8 @@ func TestDelete(t *testing.T) {
 		// FIXME: cannot check if it's VCH or not
 		// testNewVCHFromCompute(input.ComputeResourcePath, input.DisplayName, validator, t)
 		//	testDeleteVCH(validator, conf, t)
```
Comment (gigawhitlocks) [inline lib/install/management/delete_test.go]: uncomment line 92 too, it should work now :)

**Comment by gigawhitlocks** [comment]
One comment, otherwise LGTM :) :+1:

File: lib/install/management/delete_test.go (hunk 1 above)
Comment (dougm) [inline lib/install/management/delete_test.go]: Looks like we need to implement SearchIndex in vcsim for that to work, so I'll leave it for now.

**Pull request merged**